tests/unit/h/services/group_members_test.py

```diff
@@ -36,27 +37,25 @@ def test_it_removes_user_from_group(
         self, group_members_service, factories, creator
     ):
         group = factories.Group(creator=creator)
-        new_member = factories.User()
-        group.members.append(new_member)
+        new_member = factories.User(memberships=[GroupMembership(group=group)])
 
         group_members_service.member_leave(group, new_member.userid)
 
         assert new_member not in group.members
 
-    def test_it_is_idempotent(self, group_members_service, factories, creator):
```

Comment:
This test was a bit odd and (with the new implementation of `member_leave()`) it no longer tests what it's supposed to: if you just call `member_leave()` twice without committing the transaction the user will still be a member the second time. Rewrote it as a slightly different test